src/main/java/repository/DBRepository.java

```diff
@@ -71,4 +75,218 @@ private DBRepository() throws SQLException, ClassNotFoundException, IOException
+     * @throws SQLException if an SQL error occurs
+     */
+    private void insertBooking(Group group, Time startTime, Time endTime, String room) throws SQLException {
+        String query = "INSERT INTO Booking (Reservation, StartTime, EndTime, Room) VALUES (?, ?, ?, ?);";
+        PreparedStatement statement = connection.prepareStatement(query);
+
+        statement.setInt(1,group.getGroupNumber());
+        statement.setTime(2,startTime);
+        statement.setTime(3,endTime);
+        statement.setString(4,room);
+
+        statement.execute();
+    }
+
+    /**
+     * Inserts into the table Appointment, and, if applicable, into Reservation and Booking as well
+     * @param appointment the Appointment to be inserted
+     * @throws SQLException if an SQL error occurs
+     */
+    public void insertAppointment(Appointment appointment) throws SQLException {
+        String query = "INSERT INTO Appointment (Date, Activated, StartTime, EndTime, Note) VALUES (?, ?, ?, ?, ? );";
+        PreparedStatement statement=connection.prepareStatement(query);
+
+        statement.setDate(1,Date.valueOf(appointment.getDate()));
+        statement.setBoolean(2,(appointment.getState() != Appointment.State.DEACTIVATED));
+        statement.setTime(3,Time.valueOf(appointment.getTimeWindowStart()));
+        statement.setTime(4,Time.valueOf(appointment.getTimeWindowEnd()));
+        statement.setString(5,appointment.getNote());
+
+        statement.execute();
+
+        if(appointment.getState() == Appointment.State.RESERVED){
+            insertReservation(appointment.getReservation().getGroup(),Date.valueOf(appointment.getDate()));
+        }
+
+        if(appointment.getState() == Appointment.State.BOOKED){
+            insertReservation(appointment.getBooking().getGroup(),Date.valueOf(appointment.getDate()));
+            insertBooking(appointment.getBooking().getGroup(),
+                    Time.valueOf(appointment.getBooking().getStartTime()),
+                    Time.valueOf(appointment.getBooking().getEndTime()),
+                    appointment.getBooking().getRoom());
+        }
+    }
+
+    /**
+     * Inserts a single group into the database
+     * @param group the group to be inserted
+     * @throws SQLException if an SQL error occurs
+     */
+    public void insertGroup(Group group) throws SQLException {
+        String query = "INSERT INTO Groups (GroupNumber) VALUES (?);";
+        PreparedStatement statement = connection.prepareStatement(query);
+
+        statement.setInt(1,group.getGroupNumber());
+
+        statement.execute();
+    }
+
+    /**
+     * Updates an Appointment, and, if applicable, its Reservation and Booking
+     * by deleting the old data and inserting the new data
+     * @param appointment The Appointment to be updated
+     * @throws SQLException if an SQL error occurs
+     */
+    public void updateAppointment(Appointment appointment) throws SQLException {
+        deleteAppointment(appointment);
```

Comment:
Wäre die Verwenung eines SQL Updates nicht besser?

Reply:
An sich: ja. Aber da ja mit dieser Methode in den 3 Tabellen Appointment, Reservation und Booking gearbeitet wird, würde das zu sehr vielen verschachtelten Bedingungen kommen. Je nachdem, was verändert wurde, ist in Booking und Reservation ggf. ein `INSERT` oder `DELETE` statt `UPDATE` nötig, was zu sehr viel Gefummel führen würde. Daher hab ich's so gemacht.

Reply:
Ist erstmal ok